src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -385,11 +386,15 @@ public static ImmutableArray<TResult> SelectAsArray<TSource, TResult>(this IRead
             if (source == null)
                 return ImmutableArray<TResult>.Empty;
 
-            var builder = ArrayBuilder<TResult>.GetInstance(source.Count);
+            var builder = new TResult[source.Count];
```

Comment:
There are a _lot_ of `SelectAsArray` overloads, and I don't feel particularly comfortable with having one use a different pattern than the rest. If we think that this is a beneficial pattern, we should use it uniformly.

Reply:
The other ones all take in an IEnumerable, so there is no way to no the array size to allocate up front.

Reply:
I don't necessarily just mean the ones in this file. For example, all of the ones that operate on `ImmutableArray` itself use the `ArrayBuilder` pattern. I would rather see a dedicated commit that switches us to the pattern across the board.

Reply:
Went ahead and changed what I could find in the ImmutableArrayExtensions class